src/components/DashboardPage.tsx

```diff
@@ -160,25 +159,37 @@ Events.defaultProps = {
   loading: false,
 };
 
-const DashboardPage = ({ activeTab: initialActiveTab }: Props) => {
+type Props = { activeTab: TabOptions; page?: number };
+
+const DashboardPage = ({ activeTab: initialActiveTab, page }: Props) => {
   const { t } = useTranslation();
+  const { asPath } = useRouter();
+
+  const getCurrentUrl = () =>
+    new URL(`${window.location.protocol}//${window.location.host}${asPath}`);
 
   const { cookies } = useCookiesWithOptions(['user']);
   const [activeTab, setActiveTab] = useState(initialActiveTab);
-  const [currentPageItems, setCurrentPageItems] = useState(1);
+  const [currentPageItems, setCurrentPageItems] = useState(page ?? 1);
 
   const useGetItemsByCreator = useMemo(() => GetItemsByCreatorMap[activeTab], [
     activeTab,
   ]);
 
+  useEffect(() => {
+    if (currentPageItems === 1) return;
+    const url = getCurrentUrl();
+    url.searchParams.set('page', currentPageItems.toString());
+    window.history.pushState(undefined, '', url.toString());
```

Comment:
Because I don't wan't to retrigger a refresh of the page each time we switch tabs. I'll take another look if I can implement it better with Next router.

Reply:
`router.push()` does not refresh, it works like a `Link` https://nextjs.org/docs/api-reference/next/router#routerpush

Reply:
A potential problem is that router.push is a promise, you can't await a promise in a useEffect

Reply:
But `window.history.pushState()` has the problem that it doesn't update the router's internal state which will probably lead to bugs like:

- Click dashboard in the sidebar and you go to `/events`
- Click the places tab and it goes to `/places` in your browser's url bar
- Click dashboard again in the sidebar and you would expect to go back to `/events`, but you won't because the router thinks you are still on `/events`

Reply:
Oh I see now that this is about the pagination, not the tabs. But I think you will get similar bugs with the pagination then.

Reply:
So I've implemented it in this way now:

```javascript
const handleSelectTab = async (eventKey: TabOptions) => {
    const url = getCurrentUrl();
    url.pathname = eventKey;
    url.searchParams.set('page', '1');

    await router.push(url);
  };

  const setCurrentPage = async (page: number) => {
    const url = getCurrentUrl();
    url.searchParams.set('page', `${page}`);

    await router.push(url);
  };
```
The component renders it's layout based on the page that comes from the url, currentPage is no longer a state variable. But from initial testing, this approach is much slower.

Reply:
I don't have your code open here right now, but wouldn't clicking a link be slow as well then if `router.push` is slow?

Reply:
And have you tried shallow routing? https://nextjs.org/docs/routing/shallow-routing

Reply:
I have added shallow routing only for the pagination now, and it seems to be working well. Hopefully we don't run into some side effects 😄

Reply:
The caveat is that you can only do shallow routing as long as you stay on the same page. So for pagination (with a query param) it should work fine. But if you push a route change from `/events` to `/places` for example, it won't be shallow even if you ask it to.